_data-prepper/common-use-cases/anomaly-detection.md

```diff
@@ -0,0 +1,210 @@
+
+The `log-to-metrics-anomaly-detector-pipeline` sub-pipeline receives the aggregated histogram metrics from the `log-to-metrics-pipeline` sub-pipeline and sends them to the `anomaly_detector` processor to detect anomalies using the Random Cut Forest algorithm. If it detects anomalies, it sends them to an OpenSearch index named `log-metric-anomalies`.
+
+```json
+apache-log-pipeline-with-metrics:
+  source:
+    http:
+      # Provide the path for ingestion. ${pipelineName} will be replaced with pipeline name configured for this pipeline.
+      # In this case it would be "/apache-log-pipeline-with-metrics/logs". This will be the FluentBit output URI value.
+      path: "/${pipelineName}/logs"
+  processor:
+    - grok:
+        match:
+          log: [ "%{COMMONAPACHELOG_DATATYPED}" ]
+  sink:
+    - opensearch:
+        ...
+        index: "logs"
+    - pipeline:
+        name: "log-to-metrics-pipeline"
+
+log-to-metrics-pipeline:
+  source:
+    pipeline:
+      name: "apache-log-pipeline-with-metrics"
+  processor:
+    - aggregate:
+        # Specify the required identification keys
+        identification_keys: ["clientip", "request"]
+        action:
+          histogram:
+            # Specify the appropriate values for each the following fields
+            key: "bytes"
+            record_minmax: true
+            units: "bytes"
+            buckets: [0, 25000000, 50000000, 75000000, 100000000]
+        # Pick the required aggregation period
+        group_duration: "30s"
+  sink:
+    - opensearch:
+        ...
+        index: "histogram_metrics"
+    - pipeline:
+        name: "log-to-metrics-anomaly-detector-pipeline"
+
+log-to-metrics-anomaly-detector-pipeline:
+  source:
+    pipeline:
+      name: "log-to-metrics-pipeline"
+  processor:
+    - anomaly_detector:
+        # Specify the key on which to run anomaly detection
+        keys: [ "bytes" ]
+        mode:
+          random_cut_forest:
+  sink:
+    - opensearch:
+        ...
+        index: "log-metric-anomalies"
+```
+{% include copy-curl.html %}
+
+## Metrics from traces
+
+You can derive metrics from traces and find anomalies in these generated metrics. In this example, the `entry-pipeline` sub-pipeline receives trace data from the OpenTelemetry Collector and forwards it to the following sub-pipelines:
+
+- `span-pipeline` – Extracts the raw spans from the traces. It sends the raw spans to any indexes OpenSearch prefixed with `otel-v1-apm-span`.
+
+- `service-map-pipeline` – Aggregates and analyzes it to create documents that represent connections between services. It sends these documents to an OpenSearch index named `otel-v1-apm-service-map`. You can then see a visualization of the service map through the [Trace Analytics]({{site.url}}{{site.baseurl}}/observing-your-data/trace/index/) plugin for OpenSearch Dashboards.
```

Comment:
Can we replace the first instance of "it" with the noun? Aggregates and analyzes what?